functional-tests/src/test/java/com/redhat/che/functional/tests/RhCheAbstractTestClass.java

```diff
@@ -26,11 +27,18 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(RhCheAbstractTestClass.class);
 
+  public static final String CHE_PROD_PREVIEW_URL = "che.prod-preview.openshift.io";
+  public static final String CHE_PROD_URL = "che.openshift.io";
+
   @Inject private Ide ide;
   @Inject private NotificationsPopupPanel notificationsPopupPanel;
   @Inject private ProjectExplorer projectExplorer;
   @Inject private CodenvyEditor editor;
 
+  @Inject
+  @Named("che.host")
+  String cheHost;
```

Comment:
@ScrewTSW could we avoid injecting env var in the multiple places? Could we have some class that would be in charge of checking the infrastructure with 2 methods "isProd" / "isPreview" that would be injected. Currently we have multiple duplications of methods in different classes e.g. :

  public boolean isProd() {
    return CHE_PROD_URL.equals(cheHost);
  }

Reply:
I could maybe create a helper class, but in this case it has been discussed with @Katka92 and @rhopp 
This class cannot extend the Abstract test class because this one is che7 based selenium test, which uses completely different page fragments and using the AbstractTestClass would break it (it would try to `@Inject` che6 based fragments and cause test failure) hence the repeated logic.

The question is, is it really worth investing extra time and effort into something that will be deleted in 2-3 weeks?

Reply:
I find it really easy to implement e.g. https://github.com/redhat-developer/che-starter/blob/master/src/main/java/io/fabric8/che/starter/util/CheServerUrlProvider.java#L21 but up to you if this code will be obsolete and deleted in a couple of weeks